sys/include/net/gnrc.h

```diff
@@ -155,26 +155,46 @@
  * Transmitting Packets
  * --------------------
  *
- * To relay the @ref net_gnrc_pkt to interested threads the @ref
- * net_gnrc_netapi offers a dispatch functionality. The following example
+ * A packet is transmitted by relaying it to threads interested in handling (and
+ * dispatching) packets of its type. To do this, the @ref
+ * net_gnrc_netapi offers a dispatch functionality called @ref
+ * gnrc_netapi_dispatch_send().
+ *
+ * The following example
  * sketches the usage and assumes a valid @ref net_gnrc_pkt named `pkt`.
  *
  * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ {.c}
+ * gnrc_pktsnip_t *pkt;
+ *
+ * pkt = gnrc_pktbuf_add(NULL, data, size, GNRC_NETTYPE_UNDEF);
+ * if (pkt == NULL) {
+ *      puts("Error: unable to copy data to packet buffer\n");
+ *      return;
+ * }
  * if (!gnrc_netapi_dispatch_send(GNRC_NETTYPE_UDP, 80, pkt)) {
  *      puts("Error: no thread is interested");
  *      gnrc_pktbuf_release(pkt);
  *      return;
  * }
  * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  *
- * The @ref net_gnrc_pkt "pkt" will be send to all threads that registered for
- * @ref GNRC_NETTYPE_UDP and the demux context `80`. Every registered thread
+ * First, the data to be sent is added to the packet buffer. This ensures its
+ * intactness during the sending process. After the data in @data has been added
+ * to the packet buffer, its parent data structure can safely be freed or re-used.
+ *
+ * Then, the @ref net_gnrc_pkt "pkt" will be sent to all threads that registered
+ * for @ref GNRC_NETTYPE_UDP and the demux context `80`. Every registered thread
  * will receive a @ref GNRC_NETAPI_MSG_TYPE_SND command and can access the @ref
  * net_gnrc_pkt. If @ref gnrc_netapi_dispatch_send() is replaced by @ref
  * gnrc_netapi_dispatch_receive() then threads will receive the @ref
  * GNRC_NETAPI_MSG_TYPE_RCV command instead, again with access to the @ref
  * net_gnrc_pkt.
  *
+ * @note The IP and UDP headers for pkt have to be built manually before
```

Comment:
True. How about... "If the data to be sent requires extra headers to be added for successful transmission (in the example, this would be IP and UDP headers), these have to be built manually ..."?